src/app/txListener.js

```diff
@@ -0,0 +1,267 @@
+'use strict'
+var async = require('async')
+var ethJSABI = require('ethereumjs-abi')
+var ethJSUtil = require('ethereumjs-util')
+var EventManager = require('ethereum-remix').lib.EventManager
+var remix = require('ethereum-remix')
+var codeUtil = remix.util.code
+var Web3VMProvider = remix.web3.web3VMProvider
+
+/**
+  * poll web3 each 2s if web3
+  * listen on transaction executed event if VM
+  * attention: blocks returned by the event `newBlock` have slightly different json properties whether web3 or the VM is used
+  * trigger 'newBlock'
+  *
+  */
+class TxListener {
+  constructor (opt) {
+    this.event = new EventManager()
+    this._appAPI = opt.appAPI
```

Comment:
I also change the signature and constructore like this:
```js
class TxListener {
  constructor (opts = {}) { // in case the constructor is called with no arguments, because they are all optional
    this.event = new EventManager()
    this._api = opts.api
  }
  /* .... */
}
```